src/kOS.Safe/Encapsulation/StringValue.cs

```diff
@@ -145,6 +145,61 @@ public string TrimStart()
         {
             return internalString.TrimStart();
         }
+        
+        /// <summary>
+        /// A wrapper around ToScalar to handle the fact that a kOS suffix can't
+        /// handle being called with zero or one args (optional arg), but can handle
+        /// a var-args list like this:
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        public ScalarValue ToScalarVarArgsWrapper(params Structure [] args)
+        {
+            if (args.Length > 1)
+                throw new KOSArgumentMismatchException(1, args.Length, "TONUMBER must be called with zero or one argument, no more.");
+            if (args.Length == 0)
+                return ToScalar();
+            else
+            {
+                return ToScalar((ScalarValue)args[0]); // should throw error if args[0] isn't ScalarValue.
+            }
+        }
+        
+        /// <summary>
+        /// Parse the string into a number
+        /// </summary>
+        /// <param name="defaultIfError">If the string parse fails, return this value instead.  Note that if
+        /// this optional value is left off, a KOSexception will be thrown on parsing errors instead.</param>
+        /// <returns></returns>
+        public ScalarValue ToScalar(ScalarValue defaultIfError = null)
+        {
+            char[] decimalMarkers = new char[] {'.', ','}; // depends on culture
+            char[] scientificNotationMarkers = new Char[] {'e', 'E'};
```

Comment:
They could be combined together now, true.  (Originally the algorithm was more complex than what it looks like now and it necessitated looking at them separately.  I didn't notice I could combine them when I simplified it.)